tests/test_parts_manager_thinking_tags.py

```diff
@@ -0,0 +1,549 @@
+        expected_events=[
+            PartStartEvent(index=0, part=ThinkingPart('content')),
+            PartStartEvent(index=1, part=TextPart('</think>')),
+        ],
+        vendor_part_id=None,
+    ),
+    Case(
+        name='no_vendor_id_after_thinking_add_partial_closing_treated_as_text',
+        chunks=['<think>content', '</th'],
+        expected_parts=[ThinkingPart('content'), TextPart('</th')],
+        expected_events=[
+            PartStartEvent(index=0, part=ThinkingPart('content')),
+            PartStartEvent(index=1, part=TextPart('</th')),
+        ],
+        vendor_part_id=None,
+    ),
+]
+
+# Category 10: No Thinking Tags (tags treated as text)
+NO_THINKING_TAGS_CASES: list[Case] = [
+    Case(
+        name='new_part_tags_as_text_when_thinking_tags_none',
+        chunks=['<think>content</think>'],
+        expected_parts=[TextPart('<think>content</think>')],
+        expected_events=[
+            PartStartEvent(index=0, part=TextPart('<think>content</think>')),
+        ],
+        thinking_tags=None,
+    )
+]
+
+# Category 11: Buffer Management (stutter, flushed)
+BUFFER_MANAGEMENT_CASES: list[Case] = [
+    Case(
+        name='existing_text_stutter_buffer_via_replace',
+        chunks=['<thi', '<think>content</think>'],
+        expected_parts=[TextPart('<thi'), ThinkingPart('content')],
+        expected_events=[
+            PartStartEvent(index=0, part=TextPart(content='<thi')),
+            PartStartEvent(index=1, part=ThinkingPart(content='content')),
+        ],
+    ),
+    Case(
+        name='stutter_buffer_in_non_last_part_handled_as_text',
+        chunks=['<thi', 'content'],
+        expected_parts=[TextPart('<thicontent')],
+        expected_events=[
+            PartStartEvent(index=0, part=TextPart(content='<thicontent')),
+        ],
+    ),
+    Case(
+        name='existing_text_add_partial_opening_flush',
+        chunks=['hello', '<thi'],
+        expected_parts=[TextPart('hello<thi')],
+        expected_events=[
+            PartStartEvent(index=0, part=TextPart('hello')),
+            PartDeltaEvent(index=0, delta=TextPartDelta(content_delta='<thi')),
+        ],
+        expected_events_before_flushing=[
+            PartStartEvent(index=0, part=TextPart('hello')),
+        ],
+    ),
+    Case(
+        name='existing_text_stutter_via_append_non_empty_content',
+        chunks=['hello', '<thi', '<think>content</think>'],
+        expected_parts=[TextPart('hello<thi'), ThinkingPart('content')],
+        expected_events=[
+            PartStartEvent(index=0, part=TextPart('hello')),
+            PartDeltaEvent(index=0, delta=TextPartDelta(content_delta='<thi')),
+            PartStartEvent(index=1, part=ThinkingPart('content')),
+        ],
+    ),
+]
+
+# Category 12: Fake or Invalid Closing (added to content)
+FAKE_CLOSING_CASES: list[Case] = [
+    Case(
+        name='existing_thinking_fake_closing_added_to_thinking',
+        chunks=['<think>content', '</fake>'],
```

Comment:
This may be more interesting to test as `'</', 'fake>'`, as it starts looking like a closing tag but then isn't